pkg/fanal/secret/builtin-rules.go

```diff
@@ -783,4 +784,13 @@ var builtinRules = []Rule{
 		SecretGroupName: "secret",
 		Keywords:        []string{"typeform"},
 	},
+	{
+		ID:              "azure-data-factory-registration-key",
+		Category:        CategoryAzure,
+		Title:           "Azure Data Factory Self-Hosted Integration Runtime registration key",
+		Severity:        "HIGH",
+		Regex:           MustCompile(`IR@[0-9a-zA-Z-]{36}@(.*?)@`),
+		SecretGroupName: "secret",
```

Comment:
You need to use `named group` in Regex or remove this field.
example without `SecretGroupName` - https://github.com/aquasecurity/trivy/blob/908a4914c7e5e23ec1a6bd96879bb0d30173005c/pkg/fanal/secret/builtin-rules.go#L116
If you need to trim secret => use `named group`:
example with `SecretGroupName` - https://github.com/aquasecurity/trivy/blob/908a4914c7e5e23ec1a6bd96879bb0d30173005c/pkg/fanal/secret/builtin-rules.go#L98-L99